.github/workflows/e2e-test.yaml

```diff
@@ -0,0 +1,78 @@
+name: "E2E Test"
+on:
+  push:
+    branches:
+      - chore/e2e-test
+  workflow_dispatch:
+  schedule:
+    - cron: 0 22 * * *
+
+jobs:
+  e2e-test:
+    strategy:
+      fail-fast: false
+      matrix:
+        environment: [staging, production]
+        include:
+          - environment: staging
+            secret-name: AI_CORE_STAGING
+          - environment: production
+            secret-name: AI_CORE_PRODUCTION
+    name: "Build and Test"
+    runs-on: ubuntu-latest
+    env:
+      WGET_ARGS: "-qO- -S --content-on-error"
+      WGET_URL: "localhost:8080"
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v4
+      - uses: pnpm/action-setup@v3
+        with:
+          version: 9.5.0
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          cache: 'pnpm'
+      - run: pnpm i --frozen-lockfile
+      - name: Create .env file
+        env:
+          aicore: ${{ secrets[matrix.secret-name] }}
+        working-directory: ./sample-code
+        run: |
+          echo "aicore='$aicore'" > .env
+          url=$(echo "$aicore" | jq -r '.serviceurls.AI_API_URL' | sed 's|^https://||')
+          echo "Using AI Core ${{ matrix.environment }} instance on $url"
+      - name: "Start App"
+        working-directory: ./sample-code
+        run: |
+          pnpm run local &
+          timeout=15
+          while ! nc -z localhost 8080; do
+            sleep 1
+            timeout=$((timeout - 1))
+            if [ $timeout -le 0 ]; then
+              echo "Server did not start within 15 seconds."
+              exit 1
+            fi
+          done
+      - name: "Health Check"
+        run: |
+          wget "$WGET_ARGS" "$WGET_URL/health"
+      - name: "OpenAI Chat Completion"
+        run: |
+          wget "$WGET_ARGS" "$WGET_URL/llm"
+      - name: "OpenAI Text Embedding"
+        run: |
+          wget "$WGET_ARGS" "$WGET_URL/embedding"
```

Comment:
[req] As discussed, this would be great as actual tests instead of scripts.

Reply:
[Experimented a bit](https://github.com/SAP/ai-sdk-js/pull/51/commits/d0f2630fcaa9f2d2bccccdff11a962881c6ec90b), but couldn't make it work so far. I'd want to do this in a separate PR and with some pair coding, I don't really understand what the proper workspace setup is, and the Cloud SDK example is rather complex 😄